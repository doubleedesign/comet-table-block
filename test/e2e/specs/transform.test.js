/**
 * WordPress dependencies
 */
import { getEditedPostContent, createNewPost, transformBlockTo } from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	createNewFlexibleTableBlock,
	createNewCoreTableBlock,
	coreTableCellSelector,
	flexibleTableCellSelector,
	clickToggleControlWithText,
	openSidebar,
	openSidebarPanelWithTitle,
} from '../helper';

describe( 'Transform from core table block to flexible table block', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'should be transformed to flexible table block with no "Fixed width table cells" option', async () => {
		await createNewCoreTableBlock();
		const cells = await page.$$( coreTableCellSelector );
		await cells[ 0 ].click();
		await page.keyboard.type( 'Core Table Block' );
		await transformBlockTo( 'Flexible Table' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'should be transformed to flexible table block heeping header & footer section', async () => {
		await createNewCoreTableBlock();
		await openSidebar();
		await openSidebarPanelWithTitle( 'Table settings' );
		await clickToggleControlWithText( 'Header section' );
		await clickToggleControlWithText( 'Footer section' );
		await transformBlockTo( 'Flexible Table' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'should be transformed to flexible table block keeping "Fixed width table cells" option', async () => {
		await createNewCoreTableBlock( { col: 6, row: 6 } );
		await openSidebar();
		await openSidebarPanelWithTitle( 'Table settings' );
		await clickToggleControlWithText( 'Fixed width table cells' );
		await transformBlockTo( 'Flexible Table' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );

describe( 'Transform from flexible table block to core table block', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'should be transformed to core table block keeping "Fixed width table cells" option', async () => {
		await createNewFlexibleTableBlock( { col: 3, row: 6 } );
		const cells = await page.$$( flexibleTableCellSelector );
		await cells[ 0 ].click();
		await page.keyboard.type( 'Flexible Table Block' );
		await transformBlockTo( 'Table' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'should be transformed to core table block with no "Fixed width table cells" option', async () => {
		await createNewFlexibleTableBlock( { col: 6, row: 3 } );
		const cells = await page.$$( flexibleTableCellSelector );
		await cells[ 0 ].click();
		await page.keyboard.type( 'Flexible Table Block' );
		await openSidebar();
		await openSidebarPanelWithTitle( 'Table Settings' );
		await clickToggleControlWithText( 'Fixed width table cells' );
		await transformBlockTo( 'Table' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	/* eslint-disable jest/no-commented-out-tests */
	// it( 'should be transformed to core table block with no style & classes table', async () => {
	// 	await createNewFlexibleTableBlock( { col: 6, row: 3 } );
	// 	await openSidebar();
	// 	await openSidebarPanelWithTitle( 'Table Settings' );
	// 	await clickToggleControlWithText( 'Scroll on PC view' );
	// 	await clickToggleControlWithText( 'Scroll on Mobile view' );
	// 	await clickToggleControlWithText( 'Stack on mobile' );
	// 	await selectOptionFromLabel( 'Fixed control', 'header' );
	// 	await inputValueFromLabel( 'Table Width', '500px' );
	// 	await transformBlockTo( 'Table' );
	// 	expect( await getEditedPostContent() ).toMatchSnapshot();
	// } );

	// 	it( 'should be transformed to core table block with no rowspan / colspan cells', async () => {
	// 		// 結合セル、未結合セルそれぞれにテキストを入れる
	// 	} );

	// 	it( 'should be transformed to core table block keeping caption text', async () => {
	// 		// 改行入りのテキストをテストする
	// 	} );

	// 	it( 'should be transformed to core table block width no inline-style cells', async () => {} );

	// 	it( 'should be transformed to core table block width no classes cells', async () => {} );

	// it( 'should be transformed to core table block width no inline-style caption text', async () => {} );
} );
