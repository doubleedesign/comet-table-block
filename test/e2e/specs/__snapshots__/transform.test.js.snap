// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform from core table block to flexible table block should be transformed to flexible table block heeping header & footer section 1`] = `
"<!-- wp:flexible-table-block/table {\\"hasFixedLayout\\":false} -->
<figure class=\\"wp-block-flexible-table-block-table\\"><table class=\\"\\"><thead><tr><th></th><th></th></tr></thead><tbody><tr><td></td><td></td></tr><tr><td></td><td></td></tr></tbody><tfoot><tr><td></td><td></td></tr></tfoot></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Transform from core table block to flexible table block should be transformed to flexible table block keeping "Fixed width table cells" option 1`] = `
"<!-- wp:flexible-table-block/table -->
<figure class=\\"wp-block-flexible-table-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Transform from core table block to flexible table block should be transformed to flexible table block with no "Fixed width table cells" option 1`] = `
"<!-- wp:flexible-table-block/table {\\"hasFixedLayout\\":false} -->
<figure class=\\"wp-block-flexible-table-block-table\\"><table class=\\"\\"><tbody><tr><td>Core Table Block</td><td></td></tr><tr><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Transform from flexible table block to core table block should be transformed to core table block keeping "Fixed width table cells" option 1`] = `
"<!-- wp:table {\\"hasFixedLayout\\":true} -->
<figure class=\\"wp-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><td>Flexible Table Block</td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:table -->"
`;

exports[`Transform from flexible table block to core table block should be transformed to core table block keeping caption text 1`] = `
"<!-- wp:table {\\"hasFixedLayout\\":true} -->
<figure class=\\"wp-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table><figcaption>Flexible<br>Table<br>Block</figcaption></figure>
<!-- /wp:table -->"
`;

exports[`Transform from flexible table block to core table block should be transformed to core table block width no option caption text 1`] = `
"<!-- wp:table {\\"hasFixedLayout\\":true} -->
<figure class=\\"wp-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table><figcaption>Flexible Table Block</figcaption></figure>
<!-- /wp:table -->"
`;

exports[`Transform from flexible table block to core table block should be transformed to core table block width no style & class cells 1`] = `
"<!-- wp:table {\\"hasFixedLayout\\":true} -->
<figure class=\\"wp-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><th>Flexible Table Block</th><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:table -->"
`;

exports[`Transform from flexible table block to core table block should be transformed to core table block with no "Fixed width table cells" option 1`] = `
"<!-- wp:table {\\"hasFixedLayout\\":true} -->
<figure class=\\"wp-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><td>Flexible Table Block</td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:table -->"
`;

exports[`Transform from flexible table block to core table block should be transformed to core table block with no rowspan / colspan cells 1`] = `
"<!-- wp:table {\\"hasFixedLayout\\":true} -->
<figure class=\\"wp-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><td>Cell 1</td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:table -->"
`;

exports[`Transform from flexible table block to core table block should be transformed to core table block with no style & classe table 1`] = `
"<!-- wp:table {\\"hasFixedLayout\\":true} -->
<figure class=\\"wp-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:table -->"
`;
